client-spark/spark3/src/main/java/org/apache/spark/shuffle/writer/RssShuffleWriter.java

```diff
@@ -523,20 +532,31 @@ private void resendFailedBlocks(Set<TrackingBlockStatus> failedBlockStatusSet) {
 
     for (Map.Entry<ShuffleServerInfo, List<TrackingBlockStatus>> entry :
         faultyServerToPartitions.entrySet()) {
-      Set<Integer> partitionIds =
-          entry.getValue().stream()
-              .map(x -> x.getShuffleBlockInfo().getPartitionId())
-              .collect(Collectors.toSet());
-      ShuffleServerInfo replacement = replacementShuffleServers.get(entry.getKey().getId());
-      if (replacement == null) {
-        // todo: merge multiple requests into one.
-        replacement = reassignFaultyShuffleServer(partitionIds, entry.getKey().getId());
-        replacementShuffleServers.put(entry.getKey().getId(), replacement);
+      ShuffleServerInfo faultyServer = entry.getKey();
+      List<TrackingBlockStatus> blocks = entry.getValue();
+
+      if (!taskAttemptAssignment.isReassigned(faultyServer.getId())) {
```

Comment:
This may not a good condition for triggering the request. Could you trigger multiple times if we have multiple faulty servers.

Reply:
I can't get your point yet.

Reply:
I hesitate that we should coupled the logic between faulty server and rebalance.
If a server is  a fault server, we should change the server. If  a server has a huge partition, we should use another shuffle server or split partial tasks to write another server?

Reply:
Yes. Currently, the faultyServer is marked as the above 2 meanings. Decouple may be not meaningful, because these cases have the same solution by the reassign. But the renaming is OK .

Reply:
I feel that faulty server reassignment and rebalance reassignment shouldn't be couped.
Should we reuse the code of huge partition. Maybe we should use a new code.